packages/react-server/src/ReactFlightServer.js

```diff
@@ -320,6 +326,17 @@ function serializeThenable(request: Request, thenable: Thenable<any>): number {
   return newTask.id;
 }
 
+export function serializeDirective(request: Request, directive: Directive) {
+  emitDirectiveChunk(request, directive);
+
+  if (request.pingedTasks.length === 0) {
+    // There is no already scheduled work so we need to schedule it.
+    // There are no tasks but at the end of perform work all queues will
+    // be emptied
+  }
```

Comment:
This if doesn't do anything but the fundamental issue here is that we can no longer assume that we only schedule work at the first task added.

That assumption is wrong here and in pingTask now since we might have already scheduled work.

We likely need some other flag to avoid double scheduling.

Reply:
Yeah I was thinking about making a noop task but didn't want to branch performWork. The double scheduling is bad for perf when it happens but not logically wrong right?